cmd/http-forwarder/main.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	kitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/broker"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/transformers/senml"
+	"github.com/mainflux/mainflux/writers"
+	"github.com/mainflux/mainflux/writers/api"
+	"github.com/mainflux/mainflux/writers/httpforwarder"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	svcName = "http-forwarder"
+
+	defNatsURL         = "nats://localhost:4222"
+	defLogLevel        = "error"
+	defPort            = "8990"
+	defURL             = "http://localhost:9000"
+	defSubjectsCfgPath = "/config/subjects.toml"
+	defContentType     = "application/senml+json"
+
+	envNatsURL         = "MF_NATS_URL"
+	envLogLevel        = "MF_HTTP_FORWARDER_LOG_LEVEL"
+	envPort            = "MF_HTTP_FORWARDER_PORT"
+	envURL             = "MF_HTTP_FORWARDER_URL"
+	envSubjectsCfgPath = "MF_HTTP_FORWARDER_SUBJECTS_CONFIG"
+	envContentType     = "MF_HTTP_FORWARDER_CONTENT_TYPE"
+)
+
+type config struct {
+	natsURL         string
+	logLevel        string
+	port            string
+	url             string
+	subjectsCfgPath string
+	contentType     string
+}
+
+func main() {
+	cfg := loadConfigs()
+
+	logger, err := logger.New(os.Stdout, cfg.logLevel)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	b, err := broker.New(cfg.natsURL)
+	if err != nil {
+		logger.Error(err.Error())
+		os.Exit(1)
+	}
+	defer b.Close()
+
+	repo := httpforwarder.New(cfg.url)
+
+	counter, latency := makeMetrics()
+	repo = api.LoggingMiddleware(repo, logger)
+	repo = api.MetricsMiddleware(repo, counter, latency)
+	st := senml.New(cfg.contentType)
+	if err := writers.Start(b, repo, st, svcName, cfg.subjectsCfgPath, logger); err != nil {
+		logger.Error(fmt.Sprintf("Failed to start HTTP forwarder: %s", err))
+		os.Exit(1)
+	}
+
+	errs := make(chan error, 2)
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, syscall.SIGINT)
+		errs <- fmt.Errorf("%s", <-c)
+	}()
+
+	go startHTTPService(cfg.port, logger, errs)
+
+	err = <-errs
+	logger.Error(fmt.Sprintf("HTTP forwarder service terminated: %s", err))
+}
+
+func loadConfigs() config {
+	cfg := config{
+		natsURL:         mainflux.Env(envNatsURL, defNatsURL),
+		logLevel:        mainflux.Env(envLogLevel, defLogLevel),
+		port:            mainflux.Env(envPort, defPort),
+		url:             mainflux.Env(envURL, defURL),
+		subjectsCfgPath: mainflux.Env(envSubjectsCfgPath, defSubjectsCfgPath),
+		contentType:     mainflux.Env(envContentType, defContentType),
+	}
+
+	return cfg
+}
+
+func makeMetrics() (*kitprometheus.Counter, *kitprometheus.Summary) {
+	counter := kitprometheus.NewCounterFrom(stdprometheus.CounterOpts{
+		Namespace: "httpforwarder",
+		Subsystem: "message_writer",
```

Comment:
Why `message_writer` here?

Reply:
Good question. Maybe `message_forwarder` it is better?